lib/src/findex/uid_and_value.dart

```diff
@@ -42,16 +46,25 @@ class UidAndValue {
   }
 
   static int serializeToList(Uint8List output, List<UidAndValue> values) {
-    log("serialize: output: $output");
-    if (output.isEmpty) {
-      throw Exception("Unable to serialize: output length value is 0");
+    log("serializeToList: output.length: ${output.length}");
+
+    // Check if output length is enough
+    int totalSize = values.fold(
+        0, (int accumulator, UidAndValue uv) => accumulator + uv.size());
+    log("serializeList: totalSize: $totalSize");
+    if (totalSize > output.length) {
+      log("Unable to serialize: output length (${output.length}) is insufficient, $totalSize bytes needed. Is the number of Entry Table correct?");
+      return 1;
     }
```

Comment:
Why this function do not throw instead of returning error code?

Reply:
you are right about line 56, we should throw an exception. Otherwise, we want the length of the serialized vector to do a retry (such as in `cloudproof_java`, if `rust` does not allocate a sufficient buffer, the correct size is return by the callback). This retry is used for synchronous code (without isolate)